apps/utils/locust_tests/services/server-load/README.md

```diff
@@ -0,0 +1,74 @@
+# `bfd-run-server-load`/`bfd-server-load` Service
+
+## Service Diagram
+
+![Service Deployment Diagram](docs/resources/server-load.svg)
+
+## Service Overview
+
+This sub-directory contains the source code for the `bfd-run-server-load` Jenkins job that is
+capable of running many different load testing scenarios, such as:
+
+- Load tests that ramp up infinitely until a scaling event
+  - The number of instances in the scaling group to target is configurable
+- Static load tests that run with a set number of users for a set period of time ignoring scaling
+  events
+- And more...
+
+The flexibility allows for an engineer to modify parameters and see how the BFD Server performs
+under the load conditions provided. In all, the following parameters can be modified:
+
+- SDLC Environment
+- The host to test
+- The number of _initial_ worker **nodes**/Lambdas (_not_ simulated users) to start with before the
+  Locust master starts
+- The amount of time to wait between spawning nodes/Lambdas
+- The maximum number of nodes/Lambdas to spawn
+- The maximum number of _simulated users_ for **Locust** to spawn
+  - This controls the load actually generated on the target
+  - In general, a ratio of 10:1 (10 users to 1 node/lambda) should be followed for the best
+    performance
+- The spawn rate of _simulated users_ in number of users per second
+- The test runtime limit
+  - This is used as a safeguard to ensure that runaway load testing does not occur
+  - Separate from the Lambda runtime limit
+- The amount of time to "coast" after a scaling event was encountered
+  - Allows for testing the performance during and immediately after scaling
+  - Set to `0` if the amount of load required to scale (and nothing more) is desired
+- The target number of instances that have been scaled up to before load testing stops
+- Whether to stop when a scaling event is encountered
+- Whether to stop when the node/Lambda limit is reached or not
+  - Use this as a safeguard against runway Lambda spawning during non-static load tests
+
+By default, _for the Jenkins job_, these parameters are set to values that will continue to ramp up
+the test until a scaling event is encountered at which point the test will "coast" for 10 seconds
+and stop.
+
+Note that builds/runs of this pipeline _do not_ communicate with the controller directly. The
+operator is required to indicate when the load test has finished (or allow the built-in timeout to
+lapse).
```

Comment:
Much like when we get failures in the regression suite during deployment, once the controller instance has been deployed during a load test Jenkins run the pipeline will await the operator to click either `Proceed` or `Abort` to indicate that the load test is done (those buttons can be seen either in log output or in the overall stage view). Whoever is running the Jenkins job will then monitor the controller logs via CloudWatch or by SSHing into the controller and `tail`ing the `server-load-controller.log` until the load test is finished, and then once the load test is finished the operator can go back to the Jenkins pipeline and click one of those two buttons to cleanup.

Basically, it's on the operator to watch the logs generated by the load test run to see when it finishes. If, for whatever reason, they don't happen to be monitoring the load test run it will timeout on its own 10 minutes after the load test runtime limit.